toolchain/check/class.cpp

```diff
@@ -201,6 +201,17 @@ static auto BuildVtable(Context& context, Parse::ClassDefinitionId node_id,
         derived_vtable_entry_id = build_specific_function(*i);
         override_fn.virtual_index = vtable.size();
         CARBON_CHECK(override_fn.virtual_index == fn.virtual_index);
+      } else if (auto base_vtable_entry =
+                     context.sem_ir().insts().TryGetAs<SemIR::SpecificFunction>(
+                         base_vtable_entry_id)) {
+        if (class_generic_id.has_value()) {
```

Comment:
If we drop the if condition here, does the semir change at all? GetOrAddInst seems like it would just return the existing constant value SpecificFunction if it could?